crates/sdk/src/lib.rs

```diff
@@ -148,6 +178,8 @@ impl Sdk {
         VC::Periphery: Chip<SC>,
     {
         let app_prover = AppProver::new(app_pk.app_vm_pk.clone(), app_committed_exe);
+        #[cfg(feature = "memory_profiler")]
+        self.profiler.update_memory_usage(Method::Proof);
```

Comment:
the issue is this only gets the memory usage at this point in time, but I think isn't able to capture peak memory usage during the overall function span?